app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import androidx.test.espresso.Espresso.pressBack
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+class SmokeTest {
+    private val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    private lateinit var mockWebServer: MockWebServer
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule()
+
+    @Before
+    fun setUp() {
+        mockWebServer = MockWebServer().apply {
+            setDispatcher(AndroidAssetDispatcher())
+            start()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun verifyTheNavToolBarButtonsTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        val defaultWebPageTitle = "Test_Page_1"
+
+        // Visit a URL, wait until it's loaded
+        homeScreen {
+            navigationToolbar {
+            }.enterURLAndEnterToBrowser(defaultWebPage.url) {
+                verifyPageContent(defaultWebPage.content)
+
+                // Verify the items of Nav URL Bar
+                verifyNavURLBarItems()
+
+                // Verify 'x' button actions
+                verifyXButtonActions()
+                pressBack()
+                TestAssetHelper.waitingTime
+
+                // Verify various items after returning back to the initial WebPage
+                verifyNavURLBar()
+                verifyUrl((defaultWebPage.url).toString())

```

Comment:
oh, now test is failing here, it does not find the  urlBar element...it is like nothing in that page is seen when running the test...what about tryiing  to add aftter line 58 this:
` browserScreen {
                    verifyNavURLBar()
                    verifyUrl((defaultWebPage.url).toString())
                }`

Kind of forcing to say where you are.. and then continue with `.openHomeScreen{....}` at least locally that works...worth trying...

Reply:
![image](https://user-images.githubusercontent.com/46127128/82916133-42ea1c00-9f8f-11ea-9dfb-7db38bca5211.png)

Is the the one the approach you were suggesting or should I need to add any other?

Reply:
that's right but change the name, is not openBrowserScreen but just browserScreen

Reply:
> that's right but change the name, is not openBrowserScreen but just browserScreen

There isn't any transition named browserScreen , I think I need to create one should I go ahead with that one? Sorry for so many follow ups :/

Reply:
it is not a transition, it is a function in BrowserRobot